api/turing/service/router_deployment_service.go

```diff
@@ -349,12 +357,26 @@ func (ds *deploymentService) createServices(
 
 	// Enricher
 	if routerVersion.Enricher != nil {
+		var currEnricherReplicas *int
+		if currRouterVersion != nil &&
+			currRouterVersion.Status == models.RouterVersionStatusDeployed &&
+			currRouterVersion.Enricher != nil {
+			currEnricherSvc, err := ds.svcBuilder.NewEnricherService(
+				currRouterVersion, project, secretName,
+				knativeQueueProxyResourcePercentage, userContainerLimitRequestFactor, nil,
+			)
+			if err != nil {
+				return services, fmt.Errorf("Unable to build the details for the currently deployed enricher: %w", err)
+			}
+			currReplicas, err := controller.GetKnativeServiceDesiredReplicas(ctx,
+				currEnricherSvc.Name, currEnricherSvc.Namespace)
```

Comment:
Can we avoid creating this currEnricherSvc object and directly used the GetComponentName() and GetNamespace() function? Because, if I understand it correctly, we don't need other field from NewEnricherService().

Reply:
Fair point. I've refactored this part.